[Documentation] Remove all references to Diem, DPN and DIP.

JoshLind opened the pull request:
## Motivation

This PR replaces all references to `Diem`,`DPN` and `DIP` in the `aptos-core/documentation` directory. See the various commits for the individual steps. In addition, the PR removes the directory: `aptos-core/documentation/specifications/transactions`.

Note: all references to DIPs are redirected here: https://github.com/aptos-labs/aip

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Manually inspected all of the change sites.

## Related PRs

None.

davidiw commented:
APN -> Aptos blockchain or something equivalent might even be Aptos framework

davidiw commented on documentation/specifications/move_adapter/README.md:
```diff
@@ -263,7 +263,7 @@ gas currency. This can fail with a status code of
 `CURRENCY_INFO_DOES_NOT_EXIST` if the exchange rate cannot be retrieved.
 
 * If the transaction payload is a `ScriptFunction`, check if the on-chain
-Diem Version number is 2 or later. For version 1, validation will fail with
+Aptos Version number is 2 or later. For version 1, validation will fail with
 a `FEATURE_UNDER_GATING` status code.
```
this line is just wrong since there's no Diem version any more and ScriptFunction is everywhere

davidiw commented on documentation/specifications/move_adapter/README.md:
```diff
@@ -691,12 +691,12 @@ input transactions are executed.
 After re-validating a user transaction, the adapter processes the payload,
 depending on its contents.
 
-Diem version 1 had a fixed set of script transactions for general user
```
Move Adapter is probably a huge mess

davidiw commented on documentation/specifications/transactions/offchain-api/README.md:
```diff
@@ -1,5 +1,5 @@
-# Diem Off-Chain Protocol
```
let's delete almost everything under transactions

davidiw commented on documentation/tech-papers/lbft-reconfig/lbft-reconfig.tex:
```diff
@@ -36,8 +36,8 @@
 \input{setup}
 
 \begin{document}
-\title{Reconfiguration of the Diem Blockchain by Self-Reflecting Transactions}
```
This may very well should be removed

davidiw replied:
@zekun000 ?

zekun000 replied:
why?

davidiw replied:
no point in keeping papers for something that no longer exists

zekun000 replied:
what do you mean no longer exists? reconfiguration is still there? this still describes how it works

davidiw replied:
Eee, my bad, I felt I had read a different title. Indeed, this is still very relevant.

davidiw reviewed (commented):
nice

JoshLind commented:
@davidiw, FYI: my approach here is to rename for now. To remove any issues. Whether or not the content continues to make sense is a different matter 😄 I'd prefer to follow-up with additional PRs to remove irrelevant things. Thoughts?

gregnazario commented:
> @davidiw, FYI: my approach here is to rename for now. To remove any issues. Whether or not the content continues to make sense is a different matter 😄 I'd prefer to follow-up with additional PRs to remove irrelevant things. Thoughts?

We should work quickly to get rid of irrelevant things, because that's how I found we have thousands of places where sentences ended in `for diem` or `on diem` during the last rename.

davidiw commented:
I'd rather we delete things and make a definitive move and not a "let's get back to it". If it mattered to someone, they can resurrect it in the right way and form. We are using Git after all :).

The reality is that the changes we make today, will likely be the last changes for the next 6+ months -- if ever.

JoshLind commented:
@gregnazario, @davidiw. my initial impressions were that this needed to happen quickly. Hence the imprecision on my part. I've since been convinced that it's okay to slow it down and be more diligent in our approach. If things change, we can pivot to m0ve fast 😄

JoshLind commented on documentation/specifications/transactions/onchain/README.md:
```diff
@@ -1,11 +0,0 @@
-# On-Chain Data and Transactions
-
-Aptos transactions mutate and create state (or resources) within the set of [on-chain modules](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/framework/core/sources), primarily the [Aptos Account](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/doc/AptosAccount.md). The transaction format is defined in the [Move Adapter Specification](https://github.com/apto ...
```
This appears to be the only thing of value in the `aptos-core/documentation/specifications/transactions/*` directory and even then, it's out of date and incorrect. I think our new developer documents cover this in a better way.

JoshLind commented:
@davidiw, @gregnazario:
1. I've updated `APN` -> `Aptos Blockchain` (there's only one occurrence).
2. I've added another commit that removes the entire `aptos-core/documentation/specifications/transaction` directory. I don't think there's anything of value in there anymore. Feel free to point it out though.
3. The move adapter needs a complete revamp. Shall we follow-up once we've cleaned up the framework stuff?
4. Do we want to remove the tech papers? They're quite out of date now. Personally, I still they they have some value, but I'm torn. @zekun000?

gregnazario commented:
> @davidiw, @gregnazario:
> 
> 1. I've updated `APN` -> `Aptos Blockchain` (there's only one occurrence).
> 2. I've added another commit that removes the entire `aptos-core/documentation/specifications/transaction` directory. I don't think there's anything of value in there anymore. Feel free to point it out though.
> 3. The move adapter needs a complete revamp. Shall we follow-up once we've cleaned up the framework stuff?
> 4. Do we want to remove the tech papers? They're quite out of date now. Personally, I still think they they have some value, but I'm torn. @zekun000?

I'm of the thought that the specifications are so out of date we could probably drop them all.

zekun000 commented:
we should keep the tech paper somewhere for reference, I don't want to rewrite them again

davidiw commented:
> we should keep the tech paper somewhere for reference, I don't want to rewrite them again

git history ;)
we need to update it, we shouldn't keep a paper discussing a non-existent technology -- maybe even fork it into its own repo

davidiw commented:
I'd be fine with deleting things and coming back to this version and resurrecting them.

davidiw commented on documentation/specifications/README.md:
```diff
@@ -1,29 +1,29 @@
 # Protocol Specification
 
-This document describes the protocol specifications for the Diem Payment Network (LPN). The intended audience for this document are as follows:
+This document describes the protocol specifications for the Aptos Payment Network (LPN). The intended audience for this document are as follows:
```
so weird Aptos PN and LPN ... Aptos blockchain

JoshLind replied:
Ughh.. yeah, there's gonna be a bunch of these stragglers. I can try and do some grepping today to find missed patters and/or references to Libra :P

davidiw reviewed (approved):
unblocking, but we should strongly consider deleting the whitepapers of a project that no longer exists (LibraBFT)

JoshLind commented:
> unblocking, but we should strongly consider deleting the whitepapers of a project that no longer exists (LibraBFT)

Thanks @davidiw, agreed. We can follow up in another PR and discuss. I think some specifications are also now irrelevant.

/land
